common/execute.go

```diff
@@ -6,19 +6,56 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/briandowns/spinner"
 )
 
+func startAndWaitCommand(cmd *exec.Cmd, wg *sync.WaitGroup, err *error) {
+	defer wg.Done()
+
+	if *err = cmd.Start(); *err != nil {
+		return
+	}
+
+	if *err = cmd.Wait(); *err != nil {
+		return
+	}
+
+	err = nil
+}
+
 // RunCommand runs specified command and returns an error
 // If showOutput is set to true, command output is shown
+// Else spinner is shown while command is running
 func RunCommand(cmd *exec.Cmd, dir string, showOutput bool) error {
+	var err error
+	var wg sync.WaitGroup
+
 	cmd.Dir = dir
 
 	if showOutput {
 		cmd.Stdout = os.Stdout
 		cmd.Stderr = os.Stderr
 	}
 
-	return cmd.Run()
+	var s *spinner.Spinner
+
+	if !showOutput {
```

Comment:
That about complete generalization of such things

Potentially you could have several running processes simultaneously. And I think that all commands could be running in separate goroutine.

And addition of such thing allows you run scheduler that run several tasks. In your case "run commands" and "spinner". I think such approach makes your code better. Without ugly pointers and several IFs